File: server/game/dictionary.js
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const DICT_PATH = path.join(__dirname, '..', 'words', 'dictionary.txt');

const words = fs.readFileSync(DICT_PATH, 'utf8').split('\n').filter(Boolean);
export const DICTIONARY = new Set(words);

export function isValidWord(word) {
  return DICTIONARY.has(word.toUpperCase());
}

console.log(`[dictionary] loaded ${DICTIONARY.size} words`);

File: server/game/dictionary.py
from pathlib import Path

_DICT_PATH = Path(__file__).resolve().parent.parent / 'words' / 'dictionary.txt'

with _DICT_PATH.open('r', encoding='utf-8') as _f:
    DICTIONARY = {line.strip() for line in _f if line.strip()}


def is_valid_word(word):
    return word.upper() in DICTIONARY


print(f'[dictionary] loaded {len(DICTIONARY)} words')
